src/conn/conn_compact.c

```diff
@@ -566,7 +566,9 @@ __background_compact_server(void *arg)
                 full_iteration = false;
                 WT_ERR(__wt_buf_set(session, uri, WT_BACKGROUND_COMPACT_URI_PREFIX,
                   strlen(WT_BACKGROUND_COMPACT_URI_PREFIX) + 1));
-                __background_compact_list_cleanup(session, BACKGROUND_COMPACT_CLEANUP_STALE_STAT);
+                __background_compact_list_cleanup(session,
+                  conn->background_compact.run_once ? BACKGROUND_COMPACT_CLEANUP_OFF :
+                                                      BACKGROUND_COMPACT_CLEANUP_STALE_STAT);
```

Comment:
This comment is not related to the code change, but while reviewing this part of the code, I didn't find any use of `BACKGROUND_COMPACT_CLEANUP_STALE_STAT` in the code. Do we need this enum?

Reply:
It is used here only.
If the thread is configured to run once, we need to clean all stats once we're done.
If not, we only need to clean the stale stats at the end of every full iteration.

Reply:
Ok, I understand. There is no need for a specific code for this option, but we have to pass an option.

Reply:
@kommiharibabu, I am unresolving the thread just to make sure things are ok.

We don't use `BACKGROUND_COMPACT_CLEANUP_STALE_STAT` explicitly, but we use the two other values in [here](https://github.com/wiredtiger/wiredtiger/blob/896d8765fbf20ed82dd0bd607f6f04c905277222/src/conn/conn_compact.c#L428-L430) and if it's not the two other values, we know it's that one. I am open to changing the code as it seems to raise questions.

cc @Sean04